### PR TITLE
Tweaking the doxygen.

### PR DESCRIPTION
**Pull request by lemire**
get and is are important functions and I think we can generate better doxygen for them.

### CODE REVIEW

File: include/simdjson/document.h [hunk 1]
```diff
@@ -389,25 +389,29 @@ class element : protected internal::tape_ref {
...
-   * Tell whether the value can be cast to the given primitive type.
+   * Tell whether the value can be cast to provided type (T).
    *
    * Supported types:
    * - Boolean: bool
    * - Number: double, uint64_t, int64_t
    * - String: std::string_view, const char *
-   * - Array: array
+   * - Array: dom::array
+   * - Object: dom::array
```
Comment (jkeiser) [inline include/simdjson/document.h:399]: dom::object (same below).

Comment (lemire) [reply include/simdjson/document.h:399]: Thanks for catching this typo. Will fix.

Comment (lemire) [reply include/simdjson/document.h]: Fixed.

**Comment by lemire** [comment]
@jkeiser I think you need to approve the update.

**Comment by jkeiser** [comment]
Oh! I think I'll start using Approve for minor changes like this where there's likely no need to recheck :)

**Pull request merged**